test/CSharpPrecompiledTestProjects/MultipleDependencyVersions/MultipleDependencyVersions.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+
+namespace MultipleDependencyVersions
+{
+    public static class MultipleDependencyVersions
+    {
+        [FunctionName("MultipleDependencyVersions")]
+        public static IActionResult Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req)
+        {
+            // The host uses version 5.5 of this assembly, but the deps.json says that the app
+            // should use 5.6.
+            Type t55 = Dependency55.TypeGetter.ReturnSecurityKeyType();
+            Type t56 = Dependency56.TypeGetter.ReturnSecurityKeyType();
+
+            if (!Equals(t55, t56) || !Equals(t55.Assembly, t56.Assembly))
+            {
+                throw new InvalidOperationException();
+            }
+
+            return new OkResult();
```

Comment:
This failed before the change as the first call would load the host's version (5.5) and the second would load 5.6